openassessment/workflow_batch_update_api.py

```diff
@@ -0,0 +1,629 @@
+"""
+Provides functionality to batch update ORA workflows for different scopes
+"""
+
+import logging
+import time
+import datetime
+from django.utils import timezone
+from celery import shared_task
+
+from opaque_keys.edx.keys import UsageKey
+from openassessment.runtime_imports.functions import modulestore
+from openassessment.assessment.models import PeerWorkflow
+from openassessment.workflow import api
+
+logger = logging.getLogger(__name__)
+
+
+@shared_task(bind=True,
+             acks_late=True,
+             autoretry_for=(Exception,),
+             max_retries=3,
+             retry_backoff=True,
+             retry_backoff_max=500,
+             retry_jitter=True)
+def update_workflows_for_all_blocked_submissions_task(self):  # pylint: disable=unused-argument
+    """
+    Async task wrapper
+    """
+    return update_workflows_for_all_blocked_submissions()
+
+
+@shared_task(bind=True,
+             acks_late=True,
+             autoretry_for=(Exception,),
+             max_retries=3,
+             retry_backoff=True,
+             retry_backoff_max=500,
+             retry_jitter=True)
+def update_workflows_for_course_task(self, course_id):  # pylint: disable=unused-argument
+    """
+    Async task wrapper
+    """
+    return update_workflows_for_course(course_id)
+
+
+@shared_task(bind=True,
+             acks_late=True,
+             autoretry_for=(Exception,),
+             max_retries=3,
+             retry_backoff=True,
+             retry_backoff_max=500,
+             retry_jitter=True)
+def update_workflows_for_ora_block_task(self, item_id):  # pylint: disable=unused-argument
+    """
+    Async task wrapper
+    """
+    return update_workflows_for_ora_block(item_id)
+
+
+@shared_task(bind=True,
+             acks_late=True,
+             autoretry_for=(Exception,),
+             max_retries=3,
+             retry_backoff=True,
+             retry_backoff_max=300,
+             retry_jitter=True)
+# pylint: disable=unused-argument
+def update_workflow_for_submission_task(self, submission_uuid, assessment_requirements=None, course_settings=None):
+    """
+    Async task wrapper
+    """
+    return update_workflow_for_submission(submission_uuid, assessment_requirements, course_settings)
+
+
+def logging_decorator(func):
+    """
+    Does what it says. Expects `WorkflowUpdateResult` as a decorated function return value
+    """
+
+    def wrapper(*args, **kwargs):
+        start = time.time()
+        result = func(*args, **kwargs)
+        end = time.time()
+        log_message = ""
+        if result is not None:
+            for key, value in result.__dict__.items():
+                log_message = log_message + " " + str(key) + "=" + str(value)
+        log_message = log_message + " processing_time=" + str(round(end - start, 5))
```

Comment:
Nit: here is a good place for an f-string, also probably don't want to call `__dict__`directly.

```suggestion
            for key, value in dict(result).items():
                log_message = log_message + f" {str(key)}={str(value)}"
        log_message = log_message + f" processing_time={str(round(end - start, 5))}
```

Or even shorter:

```suggestion
        log_message = " ".join([f"{str(key)}={str(value)}" for key, value in dict(result).items()])
        log_message += f" processing_time={str(round(end-start, 5))}"
```

Reply:
updated, however result object is not a dictionary so still hacky `__dict__` access